ipv6/uint128.go

```diff
@@ -0,0 +1,207 @@
+/*
+This file takes influence from uint128: https://github.com/davidminor/uint128
+
+Copyright (c) 2014 David Minor
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+package ipv6
+
+import "math/bits"
+
+type uint128 struct {
+	high, low uint64
+}
+
+// Uint128FromUint64 returns a uint128 from its two 64 bit unsigned representations
+func Uint128FromUint64(high uint64, low uint64) uint128 {
+	return uint128{high, low}
+}
+
+// Uint128FromBytes returns the the uint128 converted from a slice of 16 bytes
+func Uint128FromBytes(s []byte) uint128 {
+	return uint128{
+		high: uint64(s[0])<<56 | uint64(s[1])<<48 | uint64(s[2])<<40 | uint64(s[3])<<32 | uint64(s[4])<<24 | uint64(s[5])<<16 | uint64(s[6])<<8 | uint64(s[7]),
+		low:  uint64(s[8])<<56 | uint64(s[9])<<48 | uint64(s[10])<<40 | uint64(s[11])<<32 | uint64(s[12])<<24 | uint64(s[13])<<16 | uint64(s[14])<<8 | uint64(s[15]),
+	}
+}
+
+// MaxUint128 is the maximum integer that can be stored in a uint128, "all ones"
+func MaxUint128() uint128 {
+	return uint128{^uint64(0), ^uint64(0)}
+}
+
+// OnesCount128 returns the number of one bits ("population count") in x.
+func OnesCount128(x uint128) int {
+	return bits.OnesCount64(x.high) + bits.OnesCount64(x.low)
+}
+
+// LeadingZeros128 returns the number of leading zero bits in x; the result is 128 for x == 0.
+func LeadingZeros128(x uint128) int {
+	leadingZeros := bits.LeadingZeros64(x.high)
+	if leadingZeros == 64 {
+		leadingZeros += bits.LeadingZeros64(x.low)
+	}
+	return leadingZeros
+}
+
+// ToBytes returns a slice representation of 16 bytes of the uint128
+func (me uint128) ToBytes() []byte {
```

Comment:
You followed my example of using `me`. lol. I might be the only one in the world that does that.

Basically, I reject the premise that you shouldn't use names like `this` or `self` here. I think the one letter names that have replaced it are worse. But, one of the tools (go vet maybe) complains if you do so I found one that it doesn't complain about. :)